tasks/upgrade.yml

```diff
@@ -44,6 +44,23 @@
     steps:
       - description: "Upgrading host: {{ item.name }}"
 
+- name: Gather self-heal facts about all gluster hosts in the cluster
+  gluster_heal_info:
+    name: "{{ volume_item.name }}"
+    status_filter: self-heal
+  register: self_heal_status
+  retries: 6
+  delay: 300
+  until: >
+     self_heal_status is defined and
+     self_heal_status.glusterfs.heal_info | map(attribute='no_of_entries') | select('defined') | list | map('int') | sum == 0
+  delegate_to: "{{ host_info.ovirt_hosts[0].address }}"
```

Comment:
If one host is healing, does it mean that all hosts in the cluster are also healing? If not wouldn't it be better to skip the host and continue with next one instead of failing the whole cluster upgrade?